### DISCUSSION TITLE
Need you help - Wizard form with barcode field

### DISCUSSION THREAD

**Discussion by Grazulex** [filamentphp/filament#3798] [Help] [answered]
Hello, 
I need your help to find the best way to create a wizard form.
On one of the step, I need a textinput and the user will "read" in this a barcode. Than I need to search in a other model if I find this record, if yes, I will show this in a table (widget) in the bottom of this form. If I don't find this record in my model, than I need to created this and when is done, show this in the same table (widget). I need to read a lot of child records (OneToMany)

My problem, is that I need to have the parent record ID to create the children but with a wizard form, the record is created when the wizard is done on the last step.
And i need this ID for my widget (table) too...I need this to make a filter via the $this-record value but I don't have this too in the create action.

Can I create a record directly after the first step ? if yes, how ?
How do you managed this barcode/text field ?? I was thinking on a select field with the serachable option...

Thanks you

**Comment by danharrin** [comment, marked as answer]
We have a `saveRelationshipsUsing()` on all form components that can be used to save the child models. The callback accepts many things such as `$record` and `$state`. You can use `dehydrated(false)` on top to prevent it from being saved as a normal database field.
